repository/repository.go

```diff
@@ -1 +1,215 @@
 package repository
+
+import (
+	"attendance/util"
+	"fmt"
+	"time"
+
+	"github.com/go-pg/pg"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+type Repository interface {
+	GetUser(username string) *User
+	AddNewUser(user *User) error
+	GetCurrentStatus(username string) (bool, []Attendance)
+	AddNewPunchIn(username string) error
+	AddNewPunchOut(username string, attendance Attendance) error
+	GetTeacherAttendance(username string, data GetTeacherAttendanceJSON) []Attendance
+	GetClassAttendance(data GetClassAttendanceJSON) []StudentAttendanceJSON
+	GetStudentAttendance(username string, data GetStudentAttendanceJSON) []Attendance
+	GetDailyStats(data GetHomeJSON) (int, int, int, int)
+}
+
+type RepositoryImpl struct {
+	db *pg.DB
+}
+
+func NewRepositoryImpl(db *pg.DB) *RepositoryImpl {
+	return &RepositoryImpl{db: db}
+}
+
+func (impl *RepositoryImpl) GetUser(username string) *User {
```

Comment:
I think even in case of no user found it will return empty struct, i also tested it that is why i didn't handled error specific to it. Everywhere when a user is not found the response returned is Status Unauthorised.